common/src/main/java/dev/latvian/mods/kubejs/block/BlockStateModifyPlacementCallbackJS.java

```diff
@@ -0,0 +1,106 @@
+package dev.latvian.mods.kubejs.block;
+
+import dev.latvian.mods.kubejs.item.ItemStackJS;
+import dev.latvian.mods.kubejs.level.BlockContainerJS;
+import dev.latvian.mods.kubejs.level.LevelJS;
+import dev.latvian.mods.kubejs.player.PlayerJS;
+import dev.latvian.mods.kubejs.util.UtilsJS;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.item.context.BlockPlaceContext;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.material.Fluid;
+import net.minecraft.world.level.material.FluidState;
+import net.minecraft.world.phys.Vec3;
+import org.jetbrains.annotations.Nullable;
+
+public class BlockStateModifyPlacementCallbackJS extends BlockStateModifyCallbackJS {
+	public final BlockPlaceContext context;
+	public final Block minecraftBlock;
+	public BlockContainerJS block;
+
+	public BlockStateModifyPlacementCallbackJS(BlockPlaceContext context, Block block) {
+		super(block.defaultBlockState());
+		this.context = context;
+		this.minecraftBlock = block;
+		this.block = new BlockContainerJS(context.getLevel(), context.getClickedPos());
+	}
+
+	public BlockPos getClickedPos() {
+		return context.getClickedPos();
+	}
+
+	public BlockContainerJS getClickedBlock() {
+		return getLevel().getBlock(context.getClickedPos());
+	}
+
+	public boolean canPlace() {
+		return context.canPlace();
+	}
+
+	public boolean replacingClickedOnBlock() {
+		return context.replacingClickedOnBlock();
+	}
+
+	public Direction getNearestLookingDirection() {
+		return context.getNearestLookingDirection();
+	}
+
+	public Direction getNearestLookingVerticalDirection() {
+		return context.getNearestLookingVerticalDirection();
+	}
+
+	public Direction[] getNearestLookingDirections() {
+		return context.getNearestLookingDirections();
+	}
+
+	public Direction getClickedFace() {
+		return context.getClickedFace();
+	}
+
+	public Vec3 getClickLocation() {
+		return context.getClickLocation();
+	}
+
+	public boolean isInside() {
+		return context.isInside();
+	}
+
+	public ItemStackJS getItemInHand() {
```

Comment:
Shouldn't this just be `getItem()` so that it beans to `item` like all other events?

Reply:
The latest commit should fix this.